synapse/types.py

```diff
@@ -751,3 +751,32 @@ def get_verify_key_from_cross_signing_key(key_info):
     # and return that one key
     for key_id, key_data in keys.items():
         return (key_id, decode_verify_key_bytes(key_id, decode_base64(key_data)))
+
+
+@attr.s(frozen=True, slots=True)
+class User:
```

Comment:
I think we probably want to call this `UserInfo` or something rather than `User`.